src/Extension/FluentExtension.php

```diff
@@ -617,6 +620,114 @@ public function onBeforeWrite()
         }
     }
 
+    /**
+     * @throws ValidationException
+     */
+    public function onAfterWrite(): void
+    {
+        $owner = $this->owner;
+        $stage = Versioned::get_stage() ?: Versioned::DRAFT;
+        $currentLocale = FluentState::singleton()->getLocale();
+
+        if (!$this->localisedCopyActive) {
+            return;
+        }
+
+        if (!$currentLocale) {
+            return;
+        }
+
+        if (!$owner->classNameChangeDetected) {
+            // ClassName did not change so we can bail out
+            return;
+        }
+
+        if (!$owner->isInDB() || !$owner->existsInLocale()) {
+            // This is just a sanity check
+            return;
+        }
+
+        if ($stage !== Versioned::DRAFT) {
+            // Only draft stage is relevant for the duplication
+            return;
+        }
+
+        // Get list of all localised instances of this model and duplicate relations if needed (if current one has it)
+        $locales = $owner->getLocaleInstances();
+
+        /** @var Locale $locale */
+        foreach ($locales as $locale) {
```

Comment:
maybe if you loop over $this->Locales() instead, you can re-use the logic in RecordLocale to simplify some of this code.

E.g. 


```php
$localisedRecord = $recordLocale->getRecord();
```

Reply:
I had a look at this but it I couldn't find a way to simplify this further - the goal is to iterate over all locales which have localised record except the current locale. I'm already using the existing functionality to search for such locales via `getLocaleInstances`. I can simplify how the data changes are made though (one write per model).

Reply:
you could put those into a condition and `continue` in that case. still less code, and you can use RecordLocale to check if the localisation exists.

Reply:
Not sure I follow - we are already looping a list of locales in which the model is localised (has localised record). I'm only skipping the current locale as I can't exclude it from the list as `getLocaleInstances` returns an `array` not an `ArrayList`.

Reply:
I think the problem I have is that getLocaleInstances() is solving the same problem that the Locales() method had already solved, so we end up with a lot of duplication of effort.

without this method, we could do this


```
foreach ($this->Locales() as $recordLocale) {
  // simpler "saved in this locale?" check
  if (!$recordLocale->IsDraft()) {
    continue;
  } 
$localisedRecord  = $recordLocale->getRecord();
// rest of code
}
```

Reply:
As mentioned below, I can tidy this up later; better to get what's already done in the code for now. :)